### ISSUE TITLE
Select: onItemSelect doesn't fire

### ISSUE THREAD

**Issue by jordymeow** [palantir/blueprint#2893]
My code has nothing fancy but the onItemSelect doesn't fire. The only way I could make it work is to use onClick in the items generated using itemRenderer. Is there something special to make the onItemSelect to work? I have read the documentation about Select 5-6 times but can't see why it wouldn't fire. I am not using any filters or anything at all, just items, itemRenderer, and onItemSelect, it couldn't be simpler.

Why isn't there a basic example on how to use Select? And maybe also in ES6, to make it clearer for most users?

**Comment by giladgray** [comment]
@jordymeow are you attaching the `handleClick` prop in your item renderer?

https://github.com/palantir/blueprint/blob/develop/packages/docs-app/src/examples/select-examples/films.tsx#L135

**Comment by giladgray** [comment]
re ES6-- typescript code _is_ just ES6 with type information that you can happily ignore. https://blueprintjs.com/docs/#blueprint/reading-the-docs.understanding-typescript

**Comment by giladgray** [comment]
the select docs present a pretty clear picture of the necessary steps. by the end of the page, each prop method has been spelled out with code samples. @jordymeow how can we make it more clear?

**Comment by jordymeow** [comment]
@giladgray Thanks for your reply :) I am using onClick instead of handleClick. In the item render, handleClick doesn't work (only onClick does) and onItemSelect doesn't work on the Select. Which is odd since I get the feeling it should by reading the documentation. But I actually got it (though I didn't try it yet). I got confused by the usage of handleClick, I didn't realize it was a props. 

About the documentation, I am not saying it is bad, and it is definitely nice if most of your users are using TypeScript. I don't, and for me, it just adds to the confusion. The perfect example would have been one that I can quickly copy/paste, try, and enhance. An example that doesn't require any more files or any piece of code, and not using any filters, but that would have of course used the handleClick.
